File: app/src/main/java/com/example/messenger/ResetPasswordViewModel.java
package com.example.messenger;


import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.google.firebase.auth.FirebaseAuth;

public class ResetPasswordViewModel extends ViewModel {

    private final MutableLiveData<Boolean> success = new MutableLiveData<>();
    private final MutableLiveData<String> error = new MutableLiveData<>();
    private final FirebaseAuth auth = FirebaseAuth.getInstance();
    ;

    public LiveData<Boolean> isSuccess() {
        return success;
    }

    public LiveData<String> getError() {
        return error;
    }

    public void resetPassword(String email) {
        auth.sendPasswordResetEmail(email)
                .addOnSuccessListener(unused -> success.setValue(true))
                .addOnFailureListener(e -> error.setValue(e.getMessage()));
    }
}

File: app/src/main/java/com/example/messenger/ResetPasswordViewModel.kt
package com.example.messenger

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.google.firebase.auth.FirebaseAuth

class ResetPasswordViewModel : ViewModel() {

    private val success: MutableLiveData<Boolean> = MutableLiveData()
    private val error: MutableLiveData<String> = MutableLiveData()
    private val auth = FirebaseAuth.getInstance()

    val isSuccess: LiveData<Boolean> get() = success
    val getError: LiveData<String> get() = error

    fun resetPassword(email: String) {
        auth.sendPasswordResetEmail(email)
                .addOnSuccessListener { success.value = true }
                .addOnFailureListener { error.value = it.message }
    }
}